### PR TITLE
Add protobuf impl arg in dockerfile and comments

### PR DESCRIPTION
**Pull request by PierrickVoulet**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,23 @@
 FROM php:7.1-apache
 
+ARG USE_C_PROTOBUF=false
```
Comment (Raibaz) [inline Dockerfile]: Why is this false by default? I thought we'd switch to using the C extension by default.

Comment (PierrickVoulet) [reply Dockerfile]: Good point. I should have detailed that the purpose of this PR is to add the build argument only: I set the default as false to not introduce any breaking changes.

The change to Protobuf C will be done by other PR(s) and changing the value to true will be part of it.

Comment (Raibaz) [reply Dockerfile]: Got it, thanks.

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+
 RUN apt-get update && apt-get install -y libxml2 zlib1g-dev git unzip
 
+# Install PHP extension(s) required for development
```
Comment (fiboknacky) [inline Dockerfile]: Nit: Please end all your code comments with a period.

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+# Install and configure Composer
 RUN curl -sS https://getcomposer.org/installer | php
 RUN mv composer.phar /usr/local/bin/composer
 
+# Install and configure the gRPC extension
 RUN pecl install grpc
 RUN echo 'extension=grpc.so' >> $PHP_INI_DIR/conf.d/grpc.ini
 
+# Install and configure the C implementation of Protobuf extension if needed
+RUN if [ "$USE_C_PROTOBUF" = "false" ]; then echo 'Use PHP implementation of Protobuf'; else echo 'Use C implementation of Protobuf'; pecl install protobuf; echo 'extension=protobuf.so' >> $PHP_INI_DIR/conf.d/protobuf.ini; fi
```
Comment (fiboknacky) [inline Dockerfile]: Nit: "Using" instead of "Use" would be a bit easy to understand when logged as a text.

File: Dockerfile (hunk 2 above)
Comment (PierrickVoulet) [reply Dockerfile]: Done

File: Dockerfile (hunk 3 above)
Comment (PierrickVoulet) [reply Dockerfile]: Done

**Pull request merged**